File: source/downloads/code/LeetCode/Java/Permutation-Sequence.java
public class Solution {
    public String getPermutation(int n, int k) {
        String ret = "";
        ArrayList<Integer> list = new ArrayList<Integer>();
    
        for(int i=1;i<=n;i++){
            list.add(i);
        }
        int factor = 1;
        // calculate n!
        for (int i=1;i<n;i++)
            factor = factor*i;
        
        k = k-1;
        for (int i=n-1;i>=0;i--){
            ret += Integer.toString(list.get(k/factor));
            list.remove(k/factor);
            k = k%factor;
            if(i!=0)
                factor = factor/i;
        
        }
        return ret;

    }
}

File: downloads/code/LeetCode/Python/Permutation-Sequence.py
class Solution(object):
   def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        #Generate the sequence
        k = k-1
        nums = [i for i in range(1,n+1)]
        factorial = 1
        # calculate (n-1)!
        for i in range(1,n):
            factorial *= i 
        ret = ""
        for i in range(n-1,-1,-1):
            ret += str(nums[k/factorial])
            nums.remove(nums[k/factorial])
            if i!=0:
                k = k%factorial
                factorial = factorial/i
        return ret

